### ISSUE TITLE
Put proper error messages when deleting records

### ISSUE THREAD

**Issue by breinjhel** [ivanceras/curtain#3]

**Comment by jmealo** [comment]
@breinjhel: Can you flesh this out?

**Comment by ivanceras** [comment]
Yep, some records that are selected for deleting are not deleted. I haven't looked into the details, but this comes down to the [filter parser](https://github.com/ivanceras/inquerest) which I am currently rewriting the library to use `nom` instead of `peg` parser.

**Comment by jmealo** [comment]
@ivanceras I'm not sure if this is relevant, but, PostgREST has a pretty nice way of passing filters/quals via query parameters. Reading the tickets and Gitter is helpful to see what real use cases look like for REST consumers. I guess you have more control here, but, there's no reason the backend for curtain couldn't be used independently of curtain. Is this outside of the scope of what you were planning?

**Comment by ivanceras** [comment]
Yes, PostgREST is relevant, in fact I even wrote it in my rest filter parser project (https://github.com/ivanceras/inquerest) as inspired by PostgREST. If I go the route of using PostgREST as the backend, then what's left will be the [front-end which is written in elm](https://github.com/ivanceras/curtain-elm). The plan is not to be limited by PostgreSQL alone, I also want to support SQLite, MySQL, etc.
